### PR TITLE
 Turn on unawaited_futures in flutter_tools

### LINKED ISSUE #10754: Re-enable unawaited_futures for flutter_tools
partially #5793

So we avoid issues like #10228 in the future resulting from unintentional races. 

Attempt to resolve as many of the lint warnings as possible before enabling.

### PR DESCRIPTION
**Pull request by xster**
Fixes #10754

### CODE REVIEW

File: packages/flutter_tools/analysis_options.yaml [hunk 1]
```diff
@@ -2,3 +2,11 @@
 # the ones from above, which include the `public_member_api_docs` rule).
 
 include: ../../analysis_options.yaml
+
+analyzer:
+  exclude:
+    - test/**
```
Comment (tvolkert) [inline packages/flutter_tools/analysis_options.yaml]: Is this still needed now that `package:test` has distinguished `expect()` from `expectLater()`?

File: packages/flutter_tools/lib/src/base/process.dart [hunk 2]
```diff
@@ -191,14 +191,17 @@ Future<int> runInteractively(List<String> command, {
...
+  // finishes.
+  process.stdin.addStream(stdin); // ignore: unawaited_futures
   // Wait for stdout and stderr to be fully processed, because process.exitCode
   // may complete first.
   await Future.wait<dynamic>(<Future<dynamic>>[
     stdout.addStream(process.stdout),
     stderr.addStream(process.stderr),
   ]);
-  return await process.exitCode;
+  final int exitCode = await process.exitCode;
```
Comment (tvolkert) [inline packages/flutter_tools/lib/src/base/process.dart]: Why not just `return await process.exitCode;` as it was before?

File: packages/flutter_tools/lib/src/commands/analyze_once.dart [hunk 3]
```diff
@@ -92,7 +92,8 @@ class AnalyzeOnce extends AnalyzeBase {
     });
 
     await server.start();
-    server.onExit.then((int exitCode) {
+    // Completing the future in the callback can't fail.
+    server.onExit.then((int exitCode) { // ignore: unawaited_futures
```
Comment (tvolkert) [inline packages/flutter_tools/lib/src/commands/analyze_once.dart]: Can the `server.onExit` future ever complete with an error?

File: packages/flutter_tools/lib/src/commands/attach.dart [hunk 4]
```diff
@@ -78,7 +78,7 @@ class AttachCommand extends FlutterCommand {
 
   @override
   Future<Null> validateCommand() async {
-    super.validateCommand();
+    await super.validateCommand();
```
Comment (tvolkert) [inline packages/flutter_tools/lib/src/commands/attach.dart]: Thank you linter 😄 (and below)

File: packages/flutter_tools/lib/src/run_hot.dart [hunk 5]
```diff
@@ -590,7 +590,7 @@ class HotRunner extends ResidentRunner {
           pause: pause
         );
         countExpectedReports += reports.length;
-        Future.wait(reports).catchError((dynamic error) {
+        await Future.wait(reports).catchError((dynamic error) {
```
Comment (tvolkert) [inline packages/flutter_tools/lib/src/run_hot.dart]: This code's hard to parse (`Future.wait(...).catchError(...).then(..., onError: ...)`), which makes it hard to immediately see if we should be waiting for this here.

Can you refactor the inlined code herein into methods to make this a little more readable?

File: packages/flutter_tools/analysis_options.yaml (hunk 1 above)
Comment (xster) [reply packages/flutter_tools/analysis_options.yaml]: Ah sure. The lints in tests are for a variety of other random reasons. Patched those as well

File: packages/flutter_tools/lib/src/base/process.dart (hunk 2 above)
Comment (xster) [reply packages/flutter_tools/lib/src/base/process.dart]: Oops, I was trying something else and forgot to revert everything

File: packages/flutter_tools/lib/src/commands/analyze_once.dart (hunk 3 above)
Comment (xster) [reply packages/flutter_tools/lib/src/commands/analyze_once.dart]: The comment was for the closure inside (in that if that closure has an error, it won't be handled by anyone (which is ok))

File: packages/flutter_tools/lib/src/run_hot.dart (hunk 5 above)
Comment (xster) [reply packages/flutter_tools/lib/src/run_hot.dart]: I think it's more that the formatting was unhelpful for readability. How is it now?

File: packages/flutter_tools/test/commands/daemon_test.dart [hunk 6]
```diff
@@ -103,9 +103,9 @@ void main() {
         notifyingLogger: notifyingLogger
       );
       commands.add(<String, dynamic>{'id': 0, 'method': 'daemon.shutdown'});
-      return daemon.onExit.then<Null>((int code) {
-        responses.close();
-        commands.close();
+      return daemon.onExit.then<Null>((int code) async {
+        await responses.close();
```
Comment (xster) [inline packages/flutter_tools/test/commands/daemon_test.dart]: @devoncarew I couldn't understand this test enough to figure out why this hangs. Do you have any clues?

Comment (xster) [reply packages/flutter_tools/test/commands/daemon_test.dart]: ok, removed

**Review by tvolkert — APPROVED** [review]
LGTM
